File: 003 - Terceiro Período [2021 - 20.2]/POO/TPS/TP2/Code/src/main/java/Models/Cliente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

/**
 *
 * @author thayn
 */


public class Cliente {
    private static int INIT_ID = 0;
    private int id;
    private String nome;
    private String endereco;
    
    public Cliente(String nome, String endereco){
        setId();
        setNome(nome);
        setEndereco(endereco);
    }

    public int getId() {
        return id;
    }

    public void setId() {
        this.id = ++INIT_ID;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }
}

File: Terceiro Período [2021 - 20.2]/POO/TPS/TP1/Code/Cliente.cpp
#include "Cliente.hpp"

static int _id = 0;

Cliente::Cliente(string nome, string endereco) : nome(nome), endereco(endereco)
{
    setId();
}

Cliente ::~Cliente(){};

void Cliente::setId()
{
    this->id = _id++;
}

void Cliente::setNome(string nome)
{
    this->nome = nome;
}

void Cliente::setEndereco(string endereco)
{
    this->endereco = endereco;
}

int Cliente::getId() const
{
    return id;
}

string Cliente::getNome() const
{
    return nome;
}

string Cliente::getEndereco() const
{
    return endereco;
}

ostream &operator<<(ostream &out, const Cliente &cliente)
{
    out << "ID: " << cliente.getId() << endl
        << "Nome: " << cliente.getNome() << endl
        << "Endereço: " << cliente.getEndereco();
    return out;
}
